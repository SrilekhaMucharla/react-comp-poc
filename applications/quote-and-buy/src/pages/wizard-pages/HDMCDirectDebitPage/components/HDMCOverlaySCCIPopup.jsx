import React from 'react';
import PropTypes from 'prop-types';
import { HDLabelRefactor } from 'hastings-components';
import { Row, Col } from 'react-bootstrap';
import {
    AnalyticsHDOverlayPopup as HDOverlayPopup
} from '../../../../web-analytics';
import * as messages from './HDMCOverlaySCCIPopup.messages';
import { getPriceWithCurrencySymbol } from '../../../../common/utils';
import { pageMetadataPropTypes } from '../../../../constant/propTypes';

const HDMCOverlaySCCIPopup = ({
    initialPayment,
    noOfPayment, costOfPayment,
    overlayButtonIcon,
    pageMetadata,
    creditAgreementData
}) => {
    return (
        <HDOverlayPopup
            webAnalyticsView={{ ...pageMetadata, page_section: messages.scciOverlayTitle }}
            webAnalyticsEvent={{ event_action: messages.scciOverlayTitle }}
            id="scci-overlay"
            className="scci_popup_overlay"
            overlayButtonIcon={overlayButtonIcon}
        >
            <Row className="overlay-header">
                <Col>
                    <HDLabelRefactor className="my-0" Tag="h2" text={messages.scciOverlayTitle} />
                </Col>
                <Col xs="auto" className="align-self-end text-center direct-debit-seci-overlay__print-col">
                    <div className="direct-debit-seci-overlay__print-col__content">
                        <i className="fa fa-print direct-debit-seci-overlay__print-col__icon" aria-hidden="true" />
                        <div className="icon-text text-regular">{messages.print}</div>
                    </div>
                </Col>
            </Row>
            <Row className="mb-3">
                <Col>
                    <HDLabelRefactor className="my-0" Tag="h5" text={messages.scciOverlaySubHeader} />
                </Col>
            </Row>
            <Row className="scci_description">
                <Col>
                    <HDLabelRefactor Tag="p" text={messages.scciHeaderDescription} />
                </Col>
            </Row>
            <Row className="scci_contact_details_header">
                <Col>
                    <HDLabelRefactor Tag="p" className="scci_info_header" text={messages.scciFirstParagraphHeader} />
                </Col>
            </Row>
            <Row className="scci_contact_details_content">
                <Col>
                    <HDLabelRefactor Tag="p" text={messages.scciFirstParagraphContent} />
                </Col>
            </Row>
            <Row className="scci_key_features_header">
                <Col>
                    <HDLabelRefactor Tag="p" className="scci_info_header" text={messages.scciSecondParagraphHeader} />
                </Col>
            </Row>
            <Row className="scci_key_features_content">
                <Col>
                    {messages.scciKeyFeaturesContent(getPriceWithCurrencySymbol({ amount: creditAgreementData.initialPayment, currency: 'gbp' }),
                        creditAgreementData.instalmentsCount - 1,
                        getPriceWithCurrencySymbol({ amount: creditAgreementData.subsequentPayment, currency: 'gbp' })).map((paragraph) => <p>{paragraph}</p>)}
                </Col>
            </Row>
            <Row className="scci_cost_of_credit_header">
                <Col>
                    <HDLabelRefactor Tag="p" className="scci_info_header" text={messages.scciThirdParagraphHeader} />
                </Col>
            </Row>
            <Row className="scci_cost_of_credit_content">
                <Col>
                    {messages.scciCreditCostContent.map((paragraph) => <p>{paragraph}</p>)}
                </Col>
            </Row>
            <Row className="scci_legal_aspects_header">
                <Col>
                    <HDLabelRefactor className="scci_info_header" Tag="p" text={messages.scciForthParagraphHeader} />
                </Col>
            </Row>
            <Row className="scci_legal_aspects_content">
                <Col>
                    {messages.scciImportantLegalAspectsContent.map((paragraph) => <p>{paragraph}</p>)}
                </Col>
            </Row>
            <Row className="scci_aditional_information_header">
                <Col>
                    <HDLabelRefactor className="scci_info_header" Tag="p" text={messages.fifthParagraphHeader} />
                </Col>
            </Row>
            <Row className="scci_aditional_information_content">
                <Col>
                    <HDLabelRefactor Tag="p" text={messages.fifthParagraphSectionA} />
                </Col>
            </Row>
            <Row className="scci_aditional_information_content">
                <Col>
                    <HDLabelRefactor Tag="p" text={messages.fifthParagraphSectionB} />
                </Col>
            </Row>
            <Row className="scci_aditional_information_content">
                <Col>
                    <HDLabelRefactor Tag="p" text={messages.fifthParagraphSectionC} />
                </Col>
            </Row>
        </HDOverlayPopup>
    );
};

HDMCOverlaySCCIPopup.propTypes = {
    initialPayment: PropTypes.number.isRequired,
    noOfPayment: PropTypes.number.isRequired,
    costOfPayment: PropTypes.number.isRequired,
    overlayButtonIcon: PropTypes.node.isRequired,
    pageMetadata: PropTypes.shape(pageMetadataPropTypes).isRequired,
    creditAgreementData: PropTypes.shape({
        amountPayable: PropTypes.number,
        aprRate: PropTypes.number,
        currencyCode: PropTypes.string,
        initialPayment: PropTypes.number,
        instalmentsCount: PropTypes.number,
        interestRate: PropTypes.number,
        subsequentPayment: PropTypes.number,
        totalPrice: PropTypes.number,
    }).isRequired,
};

export default HDMCOverlaySCCIPopup;
