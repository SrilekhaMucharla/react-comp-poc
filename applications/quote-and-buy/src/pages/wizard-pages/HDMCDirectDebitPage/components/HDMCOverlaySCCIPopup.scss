.scci_popup_overlay {
    .icon_holder_row {
        display: flex;
        flex-direction: column;
        align-items: flex-end;
    }

    .scci_info_header {
        font-weight: bold;
        margin-bottom: 14px;
    }

    .hd-overlay-btn {
        display: inline;
    }
}