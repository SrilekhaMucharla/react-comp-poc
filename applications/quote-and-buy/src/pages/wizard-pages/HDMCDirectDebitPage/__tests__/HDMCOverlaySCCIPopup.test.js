import React from 'react';

import HDMCOverlaySCCIPopup from '../components/HDMCOverlaySCCIPopup';

describe('<HDMCOverlaySCCIPopup />', () => {
    const creditAgreementData = {
        initialPayment: 5,
    };

    test('render component', () => {
        const wrapper = mount(
            <HDMCOverlaySCCIPopup creditAgreementData={creditAgreementData} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
