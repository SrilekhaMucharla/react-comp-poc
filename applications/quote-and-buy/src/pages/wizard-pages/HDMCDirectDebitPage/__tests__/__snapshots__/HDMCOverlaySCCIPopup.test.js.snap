// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDMCOverlaySCCIPopup /> render component 1`] = `
<HDMCOverlaySCCIPopup
  creditAgreementData={
    Object {
      "initialPayment": 5,
    }
  }
>
  <ForwardRef
    className="scci_popup_overlay"
    id="scci-overlay"
    onBeforeOpen={null}
    webAnalyticsEvent={
      Object {
        "event_action": "Standard Consumer Credit Information (SCCI)",
      }
    }
    webAnalyticsView={
      Object {
        "page_section": "Standard Consumer Credit Information (SCCI)",
      }
    }
  >
    <ForwardRef
      className="scci_popup_overlay"
      id="scci-overlay"
      onBeforeOpen={[Function]}
      onBlur={null}
      onCancel={null}
      onChange={null}
      onClick={null}
      onConfirm={null}
      onDelete={null}
      onEdit={null}
      onFocus={null}
      onSelect={null}
      webAnalyticsEvent={
        Object {
          "event_action": "Standard Consumer Credit Information (SCCI)",
        }
      }
    >
      <HDOverlayPopup
        cancelButton="Cancel"
        className="scci_popup_overlay"
        closeOnSelect={false}
        confirmButton="Confirm"
        customStyle=""
        disabledConfirmButton={false}
        headerBar={null}
        hidden={false}
        hideOnConfirm={true}
        id="scci-overlay"
        labelText=""
        onBeforeClose={[Function]}
        onBeforeOpen={[Function]}
        onCancel={[Function]}
        onConfirm={[Function]}
        overlayButtonIcon={null}
        overlayButtonsClassName=""
        showButtons={false}
      >
        <div
          className="hd-information-tooltip scci_popup_overlay"
        >
          <div
            className="hd-overlay-btn "
            hidden={false}
            id="scci-overlay"
            onClick={[Function]}
          />
        </div>
      </HDOverlayPopup>
    </ForwardRef>
  </ForwardRef>
</HDMCOverlaySCCIPopup>
`;
